components/script/dom/response.rs

```diff
@@ -248,11 +256,31 @@ impl ResponseMethods for Response {
         // TODO: This step relies on body and stream, which are still unimplemented.
 
         // Step 2
-        let response_response_clone = self.response.borrow().clone_for_dom_response();
         let new_response = Response::new(self.global().r());
-        *new_response.response.borrow_mut() = response_response_clone;
         new_response.Headers().set_guard(self.Headers().get_guard());
 
+        // https://fetch.spec.whatwg.org/#concept-response-clone
+        // This deviates from the spec a little, because there is no
+        // internally stored response (just top-level fields)
+        // Step 2.1
+        if *self.response_type.borrow() != DOMResponseType::Error {
+            *new_response.response_type.borrow_mut() = self.response_type.borrow().clone();
+            // TODO: How to deal with internal_response
```

Comment:
We don't need to, I don't think the fetch spec uses the internal response in the JS parts. Just mention that we skip the internal response since it's not used.

We should be unconditionally just copying the relevant fields here; no need to check for an error.